flax/linen/linear.py

```diff
@@ -50,7 +50,7 @@ def _canonicalize_tuple(x):
 class DenseGeneral(Module):
   """A linear transformation with flexible axes.
 
-    Attributes:
+    Args:
```

Comment:
I don't know about this -- why are we changing the docstring to fix RTD? The docstring was correct before -- these are attributes, not arguments (though they are arguments to the constructor but that's not what you're documenting here)

Reply:
Yes this was a way to fix it so that it appears correctly on RTD, and I did it like this because this is how JAX does it as well. I agree Attributes is more correct, but then they show up so weirdly on RTD. Looking at the example on RTD, we aren't supposed to use Args or Attributes at all! https://sphinx-rtd-theme.readthedocs.io/en/stable/_modules/test_py_module/test.html#Foo

Reply:
> By the way, RTD actually uses a totally different syntax for attributes and arguments that I think we not using at all. We could consider switching to that at some point, if we think it will work better. https://sphinx-rtd-theme.readthedocs.io/en/stable/_modules/test_py_module/test.html#Foo

Oh, just now saw this! Then yes! Let's move to this -- why not? It's better also because it co-locates the comment with the attribute. Much more useful than scrolling up and down to see the type and the name and the docstring for each attribute?

Reply:
That's also what we are using in NetKet ( see for example https://www.netket.org/docs/_generated/models/netket.models.RBM.html ).
I really suggest you using that syntax.

However it does not seem to work with attributes that are functions, such as the init_kernel arts, because RTD/Sphinx thinks they are methods instead of attributes.

If you find a workaround I'd love to hear it...

Reply:
+1 for testing on a couple of editors before changing the pydoc format.
personally, I use editor + codesearch, and rarely look at RTD, so it would be a pity if we changed something that made editor support less good for improving RTD.